stattutor/stattutor.py

```diff
@@ -0,0 +1,386 @@
+                    'score': scaled}
+            # pylint: enable=broad-except
+        return {'result': 'no-change', 'finished': self.completed,
+                'score': float(self.score)/float(self.max_problem_steps)}
+
+    def studio_view(self, dummy_context=None):
+        """ Generate the Studio page contents. """
+        html = self.resource_string("static/html/ctatstudio.html")
+        problem_dirs = [
+            '<option value="{0}"{1}>{0}</option>'.format(
+                d, ' selected' if d == self.problem else '')
+            for d in self.problems.keys()]
+        problem_dirs.sort()
+        frag = Fragment(html.format(
+            problems=''.join(problem_dirs),
+            logging='checked' if self.logtype else '',
+            logserver=self.log_url))
+        studio_js = self.resource_string("static/js/ctatstudio.js")
+        frag.add_javascript(unicode(studio_js))
+        frag.initialize_js('CTATXBlockStudio')
+        return frag
+
+    @XBlock.json_handler
+    def studio_submit(self, data, dummy_suffix=''):
+        """
+        Called when submitting the form in Studio.
+
+        Args:
+          self: the StatTutor XBlock.
+          data: a JSON object encoding the form data from
+                static/html/ctatstudio.html
+          dummy_suffix: unused but required as a XBlock.json_handler.
+        Returns:
+          A JSON object reporting the success of the operation.
+        """
+        status = 'success'
+        messages = []
+        statmodule = bleach.clean(data.get('statmodule'), strip=True)
+        logserver = bleach.clean(data.get('logserver').strip(), strip=True)
+        logging = bleach.clean(data.get('logging'), strip=True)
+        if statmodule in self.problems.keys():
+            self.problem = statmodule
+        else:
+            status = 'failure'
+            messages.append("invalid module")
+        if logging.lower() == "true":
+            self.logtype = True
+            if len(logserver) > 0:
+                self.log_url = logserver
+        else:
+            self.logtype = False
+        ret = {'result': status}
+        if len(messages) > 0:
+            ret['message'] = "; ".join(messages)
+        return ret
+
+    @XBlock.json_handler
+    def ctat_save_problem_state(self, data, dummy_suffix=''):
+        """Called from CTATLMS.saveProblemState.
+        This saves the current state of the tutor after each correct action.
+
+        Args:
+          self: the StatTutor XBlock.
+          data: A JSON object with a 'state' field with a payload of the blob
+                of 64 bit encoded data that represents the current
+                state of the tutor.
+          dummy_suffix: unused but required as a XBlock.json_handler.
+        Returns:
+          A JSON object with a 'result' field with a payload indicating the
+          success status.
+        """
+        if data.get('state') is not None:
```

Comment:
data is a JSON object not a python collection. When I tested it the python way, it gave me errors.